FrontEnd/FrontEnd.Web/Components/Pages/Recipes/Form.razor

```diff
@@ -0,0 +1,204 @@
+                                  OnClick="() => FormData.Ingredients.Add(new Ingredient())">
+                        Add new ingredient
+                    </FluentButton>
+                </FluentStack>
+                <FluentStack Orientation="Orientation.Vertical">
+                    <FluentLabel>Instructions <span style="color: red;">*</span></FluentLabel>
+                    <FluentValidationMessage For="@(() => FormData.Instructions)"/>
+                    @for (var i = 0; i < FormData.Instructions.Count; i++)
+                    {
+                        <div style="width: 100%;">
+                            @{
+                                var index = i;
+                                var instruction = FormData.Instructions[i];
+                            }
+                            <FluentStack Orientation="Orientation.Horizontal">
+                                <FluentLabel>@(index + 1).</FluentLabel>
+                                <FluentTextArea Rows="5" @bind-Value="instruction.Value" Style="width: 100%"/>
+                                <FluentButton Appearance="Appearance.Lightweight"
+                                              IconEnd="@(new Icons.Regular.Size16.Delete())"
+                                              OnClick="@(() => FormData.Instructions.Remove(instruction))"/>
+                            </FluentStack>
+                            <FluentValidationMessage For="@(() => instruction.Value)"/>
+                        </div>
+                    }
+                    <FluentButton Appearance="Appearance.Lightweight"
+                                  OnClick="() => FormData.Instructions.Add(new InstructionStep())">
+                        Add new step
+                    </FluentButton>
+                </FluentStack>
+            </FluentStack>
+        </FluentGridItem>
+        <FluentGridItem xs="12" sm="12" md="6">
+            <FluentStack Orientation="Orientation.Vertical">
+                <FluentLabel>Picture</FluentLabel>
+                <!--suppress CssUnresolvedCustomProperty -->
+                <FluentInputFile Id="picture"
+                                 Accept="image/*"
+                                 Style="height: 300px; width: 100%; border: 1px dashed var(--accent-fill-rest);">
+                    <ChildContent>
+                        <label for="picture">
+                            <FluentIcon Value="@(new @Icons.Regular.Size24.ArrowUpload())"/>
+                        </label>
+
+                        <div>
+                            Drag files here you wish to upload, or <label for="picture">browse</label> for them.
+                        </div>
+                    </ChildContent>
+                </FluentInputFile>
+            </FluentStack>
+        </FluentGridItem>
+        <FluentGridItem xs="12">
+            <FluentButton Type="ButtonType.Submit" Appearance="Appearance.Accent">Create</FluentButton>
```

Comment:
Create vs Edit. Could probably be achieved through InitialData?

Reply:
Yep